crates/primitives/src/transaction/mod.rs

```diff
@@ -1647,16 +1649,30 @@ impl Decodable for TransactionSignedEcRecovered {
 ///
 /// This is a conversion trait that'll ensure transactions received via P2P can be converted to the
 /// transaction type that the transaction pool uses.
-pub trait FromRecoveredTransaction {
+pub trait TryFromRecoveredTransaction {
+    /// The error type returned by the transaction.
+    type Error;
```

Comment:
This is basically the tryfrom crate, but I'm okay with rolling our own here, because this makes calling the functions easier to use, although perhaps we should just use TryFrom eventually